invokeai/app/services/image_storage.py

```diff
@@ -71,19 +79,75 @@ def __init__(self, output_folder: str):
                 parents=True, exist_ok=True
             )
 
+    def list(
+        self, image_type: ImageType, page: int = 0, per_page: int = 10
+    ) -> PaginatedResults[ImageResponse]:
+        dir_path = os.path.join(self.__output_folder, image_type)
+        image_paths = glob(f"{dir_path}/*.png")
+        count = len(image_paths)
+
+        # TODO: do all platforms support `getmtime`? seem to recall some do not...
+        sorted_image_paths = sorted(
+            glob(f"{dir_path}/*.png"), key=os.path.getmtime, reverse=True
+        )
+
+        page_of_image_paths = sorted_image_paths[
+            page * per_page : (page + 1) * per_page
+        ]
+
+        page_of_images: List[ImageResponse] = []
+
+        for path in page_of_image_paths:
+            filename = os.path.basename(path)
+            img = PILImage.open(path)
+            page_of_images.append(
+                ImageResponse(
+                    image_type=image_type.value,
+                    image_name=os.path.basename(path),
+                    # TODO: DiskImageStorage should not be building URLs...?
+                    image_url=f"api/v1/images/{image_type.value}/{filename}",
+                    thumbnail_url=f"api/v1/images/{image_type.value}/thumbnails/{os.path.splitext(filename)[0]}.webp",
+                    # TODO: Creation of this object should happen elsewhere, just making it fit here so it works
+                    metadata=ImageMetadata(
+                        timestamp=int(os.path.splitext(filename)[0].split("_")[-1]),
```

Comment:
Why not just use the file creation time?

Reply:
I noticed we were embedding the time stamp in the file name and figured that would be more consistent. Again this will be moot after we have a proper image mgmt. can just make it ctime